events/nats.go

```diff
@@ -297,15 +285,18 @@ func (n *NatsJetstream) subscribeAsPull(_ context.Context) error {
 	return nil
 }
 
-// PullMsg pulls upto batch count of messages from the stream through the pull based subscription.
+// XXX: the ergonomics here are weird, because we're handling potentially multiple subscriptions
```

Comment:
agreed, would it be worth having the caller specify the relative subject as a parameter,
and the method just pulls for that subject and returns?